### PR TITLE
change to modules

### PR DESCRIPTION
**Pull request by jr-nicola**
- splits the code for the plot tab, and the tables tab, into two separate modules
- put the cluster_id drop down into a separate module as well
- still has same 2 notes on R CMD check as before (next step is fixing this)

### CODE REVIEW

**Review by russHyde — CHANGES_REQUESTED** [review]
Hi, it's mostly good. The new modules really neaten it up.

I've requested a couple of changes.

- The UI/Server functions were inconsistently named,
- the `all_files` variable is independently defined in two different modules and could be a function call or argument to the modules, 
- could you add some roxygen for the new modules.

File: R/module_tables.R [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        list.files(
+          system.file("app", "www", "data", "scanner_output",
+                      cluster_choice(),
+                      package = "tfpbrowser")
+        )
+      )
+      all_files = all_files %>%
+        dplyr::mutate(filetype = sub(".*\\.", "", .data$value))
+      return(all_files)
+    })
```
[the remark this answers is not in the document]
Comment (jr-nicola) [reply R/module_tables.R:45]: I've added `get_all_files()` as a function, and called it in both modules instead

File: R/app_server.R [hunk 2]
```diff
@@ -29,174 +29,23 @@ app_server = function(input, output, session) {
...
-  # disable download button if no tables available
+  r = shiny::reactiveValues()
   shiny::observe({
-    if (table_avail() == TRUE) {
-      shinyjs::enable("download_table_button")
-    } else {
-      shinyjs::disable("download_table_button")
-    }
+    r$cluster_id = input$cluster_id
   })
```
[the remark this answers is not in the document]
Comment (jr-nicola) [reply R/app_server.R:35]: Yes - I've removed it (originally tried a different approach)

File: R/module_clusterid.R [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Module 1, which will allow to select a cluster ID
```
[the remark this answers is not in the document]
Comment (jr-nicola) [reply R/module_clusterid.R:1]: I've renamed to cluster_idUI etc.

File: R/module_clusterid.R [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+clusterid_ui = function(id) {
```
[the remark this answers is not in the document]
Comment (jr-nicola) [reply R/module_clusterid.R]: I've added comments to all modules

**Review by russHyde — APPROVED** [review]
LGTM

**Pull request merged**